drivers/pcie/host/pcie.c

```diff
@@ -216,87 +216,34 @@ bool pcie_probe_mbar(pcie_bdf_t bdf,
 	return pcie_get_mbar(bdf, reg - PCIE_CONF_BAR0, mbar);
 }
 
-/* The first bit is used to indicate whether the list of reserved interrupts
- * have been initialized based on content stored in the irq_alloc linker
- * section in ROM.
- */
-#define IRQ_LIST_INITIALIZED 0
-
-#ifndef CONFIG_MAX_IRQ_LINES
-#warning TOFIX for non-x86
-#define CONFIG_MAX_IRQ_LINES 0
-#endif
-
-static ATOMIC_DEFINE(irq_reserved, CONFIG_MAX_IRQ_LINES);
-
-static unsigned int irq_alloc(void)
-{
-	int i;
-
-	for (i = 0; i < ARRAY_SIZE(irq_reserved); i++) {
-		unsigned int fz, irq;
-
-		while ((fz = find_lsb_set(~atomic_get(&irq_reserved[i])))) {
-			irq = (fz - 1) + (i * sizeof(atomic_val_t) * 8);
-			if (irq >= CONFIG_MAX_IRQ_LINES) {
-				break;
-			}
-
-			if (!atomic_test_and_set_bit(irq_reserved, irq)) {
-				return irq;
-			}
-		}
-	}
-
-	return PCIE_CONF_INTR_IRQ_NONE;
-}
-
-static bool irq_is_reserved(unsigned int irq)
-{
-	return atomic_test_bit(irq_reserved, irq);
-}
-
-static void irq_init(void)
-{
-	extern uint8_t __irq_alloc_start[];
-	extern uint8_t __irq_alloc_end[];
-	const uint8_t *irq;
-
-	for (irq = __irq_alloc_start; irq < __irq_alloc_end; irq++) {
-		__ASSERT_NO_MSG(*irq < CONFIG_MAX_IRQ_LINES);
-		atomic_set_bit(irq_reserved, *irq);
-	}
-}
+#ifndef CONFIG_PCIE_CONTROLLER
```

Comment:
I think it'd be good to have a code comment here (or maybe in the header file) why these functions are defined only if PCIE_CONTROLLER is *not* defined. It's probably not obvious to the casual reader, and might look a bit strange.

Reply:
I made it in the header file.